gatsby-node.js

```diff
@@ -95,30 +95,36 @@ exports.createPages = async ({ graphql, actions }) => {
           return node;
         });
 
-      const hasMultiplePrograms = allPagesData.filter(node => node.type === 'pages.ProgramPage');
-
-      if (hasMultiplePrograms && hasMultiplePrograms.length > 1) {
-        // Create landing page
+      // If we spot an enterprise page, create it, and do not create programs pages
+      const isEnterprise = allPagesData && allPagesData[0].pageData.pageType === 'pages.EnterprisePage';
+      if (isEnterprise) {
         createPage({
           path: '/',
-          component: templates.programListPage,
-          context: { programs: allPagesData },
+          component: templates.enterprisePage,
+          context: allPagesData[0].pageData,
         });
-      }
-
-      if (!onlyCreateListingPage) {
-        // Create Gatsby pages for each page from portal-designer
-        allPagesData.forEach((pageData) => {
-          const isEnterprise = pageData.pageType === 'pages.EnterprisePage';
-          const template = isEnterprise ? templates.enterprisePage : templates.programPage;
-
+      } else {
+        // Create landing page if there are multiple programs pages
+        const hasMultiplePrograms = allPagesData.filter(node => node.type === 'pages.ProgramPage');
```

Comment:
nit: perhaps rename this variable to `programPages` as that line results in an array of program pages, not a boolean value like the name suggests.

And then in the `createPage` call for creating the list page, pass in the `programs` array as opposed to `allPagesData`.

```js
const programs = allPagesData.filter(node => node.type === 'pages.ProgramPage');

if (programs && programs.length > 1) {
  createPage({
    // ...
    context: { programs }
  });
}
```

And then could also use that `programs` variable for the `forEach` loop instead of `allPagesData`.

```js
programs.forEach((pageData) => {
  // ...
});
```